packages/example-app-base/src/components/EditSection/EditSection.tsx

```diff
@@ -2,101 +2,43 @@ import * as React from 'react';
 import { IEditSectionProps } from './EditSection.types';
 import { IconButton, TooltipHost } from 'office-ui-fabric-react';
 
-export class EditSection extends React.Component<IEditSectionProps, {}> {
-  public render(): JSX.Element | null {
+const readableNameMap: { [key: string]: string } = {
+  BestPractices: 'Best Practices',
+  Donts: "Don'ts"
+};
+
+/**
+ * Component for displaying an edit button next to a section header.
+ */
+export class EditSection extends React.PureComponent<IEditSectionProps> {
+  public render() {
     const { className, section, title, url } = this.props;
 
     // Check if url is falsey.
     if (!url) {
       return null;
     }
 
-    const readableSection = this._getReadableSection();
+    const readableSection = this.props.readableSection || readableNameMap[section] || section;
+    const tooltipHostId = `${title}-${section}-editButtonHost`;
 
     return (
-      this._shouldRender && (
```

Comment:
It seems reasonable to assume that as long as a URL for editing has been passed in, we should go ahead and display the edit icon (rather than doing some very elaborate checks for PageMarkdownContent children).